### PR TITLE
feat: focus markers in country

### PR DESCRIPTION
**Pull request by aplyd**
Purpose:
- focus markers in a given country when clicking the country from the list

Issues:
1. Should this feature focus the markers within the country clicked or focus the entire country? These changes focus the markers themselves and aren't aware of the country bounds.

2. What should the desired zoom level be? Currently, the map repositions to use the outermost markers as the "edge" but IMO this should be decreased (see below)
Screen Shot 2022-10-20 at 5 36 43 PM

furthermore, the map zooms in all the way if there is only a single marker in that country, which looks weird.

3. I believe we should add hover styles to the countries to further signify clicking them does something. How would you like this to look?

### CODE REVIEW

File: src/pages/timeline.tsx [hunk 1]
```diff
@@ -203,13 +184,30 @@ const Timeline: NextPage<TimelineProps> = ({ countries, ne, sw }) => {
...
+          <div
+            className={styles.country}
+            key={country}
+            onClick={() => focusMarkersInCountry(country)}
+            onKeyDown={(e) => e.key === 'Enter' && focusMarkersInCountry(country)}
+            role="button"
+            tabIndex={0}
+          >
             {flag} {name}
           </div>
```
Comment (treemmett) [inline src/pages/timeline.tsx]: This should be a button, or possibly an anchor

**Review by treemmett — CHANGES_REQUESTED** [review]
Thanks for the PR, @aplyd!

I love the idea, I wonder if the map bounding options can be used to set a padding.

As for the single marker issue, we could possibly use "set center" method instead of "set bounds".

File: src/pages/timeline.tsx (hunk 1 above)
Comment (aplyd) [reply src/pages/timeline.tsx]: Good call, I figured `role="button"` may be sufficient but using a button is definitely the better solution.

**Review by treemmett — APPROVED** [review]
Looks great, thanks for the contribution!

**Pull request merged**